google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/FakeScheduledExecutorService.java

```diff
@@ -0,0 +1,183 @@
+  }
+
+  public ScheduledFuture<?> scheduleWithFixedDelay(Runnable command, long initialDelay, long delay,
+      TimeUnit unit) {
+    throw new UnsupportedOperationException(
+        "FakeScheduledExecutorService.scheduleAtFixedRate not supported");
+  }
+
+  public void tick(long time, TimeUnit unit) {
+    List<FakeScheduledFuture> runnablesToGo = new ArrayList<>();
+    synchronized (this) {
+      clock.advance(time, unit);
+      Iterator<FakeScheduledFuture> iter = futures.iterator();
+      while (iter.hasNext()) {
+        FakeScheduledFuture scheduledFuture = iter.next();
+        if (scheduledFuture.runAtMillis <= clock.millis()) {
+          runnablesToGo.add(scheduledFuture);
+          iter.remove();
+        }
+      }
+    }
+    for (FakeScheduledFuture scheduledFuture : runnablesToGo) {
+      scheduledFuture.run();
+    }
+  }
+
+  private boolean cancel(FakeScheduledFuture toCancel) {
+    synchronized (this) {
+      Iterator<FakeScheduledFuture> iter = futures.iterator();
+      while (iter.hasNext()) {
+        FakeScheduledFuture scheduledFuture = iter.next();
+        if (scheduledFuture == toCancel) {
+          iter.remove();
+          return true;
+        }
+      }
+      return false;
+    }
+  }
+
+  private class FakeScheduledFuture implements ScheduledFuture<Object> {
+    final Callable<Object> callable;
+    final long delay;
+    final TimeUnit unit;
+    final long runAtMillis;
+
+    volatile boolean isDone = false;
+    volatile boolean isCancelled = false;
+    volatile Exception exception = null;
+    volatile Object result = null;
+
+    FakeScheduledFuture(Runnable runnable, long delay, TimeUnit unit, long runAtMillis) {
+      this.callable = Executors.callable(runnable);
+      this.delay = delay;
+      this.unit = unit;
+      this.runAtMillis = runAtMillis;
+    }
+
+    @Override
+    public long getDelay(TimeUnit requestedUnit) {
+      return unit.convert(delay, requestedUnit);
+    }
+
+    @Override
+    public int compareTo(Delayed other) {
+      return Long.compare(getDelay(TimeUnit.MILLISECONDS), other.getDelay(TimeUnit.MILLISECONDS));
+    }
+
+    @Override
+    public boolean cancel(boolean var1) {
+      if (isCancelled) {
+        return isCancelled;
+      }
+      isCancelled = FakeScheduledExecutorService.this.cancel(this);
+      return isCancelled;
+    }
+
+    @Override
+    public boolean isCancelled() {
+      return isCancelled;
+    }
+
+    @Override
+    public boolean isDone() {
+      return isDone;
+    }
+
+    @Override
+    public Object get() throws InterruptedException, ExecutionException {
+      if (!isDone()) {
+        throw new UnsupportedOperationException("FakeScheduledFuture: blocking get not supported");
+      }
+
+      if (exception != null) {
+        throw new ExecutionException(exception);
+      }
+
+      return result;
+    }
+
+    @Override
+    public Object get(long timeout, TimeUnit unit) throws InterruptedException, ExecutionException,
```

Comment:
I don't think it's ever called. But, since the main `get()` doesn't support blocking (i.e. when isDone = false), from that perspective, the timeout doesn't matter and this is a consistent implementation.